core/schemas/graph.py

```diff
@@ -119,14 +119,26 @@ def load(cls, object: dict):
         return cls(**object)
 
     @classmethod
-    def has_permissions(cls, src, target_id: str, permission: roles.Permission) -> bool:
-        vertices, paths, total = src.neighbors(
-            graph="acls", direction="outbound", max_hops=2
+    def has_permissions(
+        cls, user, target_id: str, permission: roles.Permission
+    ) -> bool:
+        acl_acl = """
+         WITH observables, entities, dfiq, indicators
+
+        FOR v, e IN 1..2 outbound @user_extended_id acls
```

Comment:
Why 2 hops instead of 1?

Reply:
The first hop will give us the users / groups who have access, the second hop is to get the users that are in the groups.